src/provider.c

```diff
@@ -607,7 +607,7 @@ static int refresh_slots(P11PROV_CTX *ctx)
         goto done;
     }
 
-    slots = OPENSSL_malloc(nslots * sizeof(struct p11prov_slot));
+    slots = OPENSSL_zalloc(nslots * sizeof(struct p11prov_slot));
```

Comment:
On second look, this might not be enough? Seems `refresh_slot_profiles` can be called again on same slot. I'll fix this also.

Reply:
Actually no code paths call it again. The name was just slightly misleading. I'll leave it as it.